executor/insert_common.go

```diff
@@ -224,9 +236,121 @@ func insertRows(ctx context.Context, base insertCommon) (err error) {
 			}
 		}
 	}
+	// Fill the batch allocated autoIDs
+	rows, err = e.autoIDAllocN(ctx, rows)
+	if err != nil {
+		return err
+	}
 	return base.exec(ctx, rows)
 }
 
+func (e *InsertValues) autoIDAllocN(ctx context.Context, rows [][]types.Datum) ([][]types.Datum, error) {
+	retryInfo := e.ctx.GetSessionVars().RetryInfo
+	length := len(e.cache)
+	for i := 0; i < length; i++ {
+		switch e.cache[i].kind {
+		case AutoIDNull:
+			// Find consecutive num.
+			var cnt int
+			var start int
+			start = i
+			for i < length {
```

Comment:
How about:
```
cnt := 1
for i+1 < length && e.cache[i+1].kind == AutoIDNull {
    cnt++
    i++
}
```
So you don't need to fix the index `i`

Reply:
Cool Done.